### PR TITLE
Feature/cd 786 customer app helm charts

### PR DESCRIPTION
**Pull request by scjohns**
- Update CKS charts
- Update CSE charts
- Add gateway charts to enable all modes

### CODE REVIEW

File: cse/values.yaml
Comment (scjohns) [inline cse/values.yaml:8]: @ehorning just thinking about what could impact Marketplace...would hardcoding docker.io here cause an issue?

Comment (scjohns) [reply cse/values.yaml:8]: Ignore, should just take out docker.io since that is default if not specified otherwise.

Comment (ehorning) [reply cse/values.yaml]: 👍 sounds good -- wouldn't cause an issue in marketplace though as that value gets overridden by the value set in the virtru-public values.yaml

Comment (scjohns) [reply cse/values.yaml]: Ah yeah okay makes sense

**Comment by scjohns** [comment]
Validated that CSE updates work with `gke-verify.sh` and `gke-deploy.sh` in virtru-public repo.

Still need to write/update readme files

File: cse/values.yaml
Comment (scjohns) [reply cse/values.yaml:109]: Discussed on Zoom a few weeks back and got on the same page. Resolving.

File: cks/templates/deployment.yaml
Comment (scjohns) [reply cks/templates/deployment.yaml:39]: Documentation will outline creating secrets separately from the helm installation. Not making changes to CSE at this time as charts are tied to marketplace. Updates for Gateway and CKS are done, documentation to be written for CKS.

**Comment by scjohns** [comment]
Re-validated CSE updates since 3 weeks ago on GKE through marketplace and all looks good.

**Comment by scjohns** [comment]
Notes RE: Secret creation...
* Conversations with @strantalis led to doing `from-file` encryption for CKS and Gateway secret creation
* CSE staying the same to not break marketplace currently

File: cks/templates/cks-keys-secret.yaml
Comment (ehorning) [inline cks/templates/cks-keys-secret.yaml:8]: why is this template still here if you're already creating the secrets above via kubectl? virtruKeys.data isn't in values.yaml so won't this fail?

File: cks/templates/configmap.yaml
Comment (ehorning) [inline cks/templates/configmap.yaml:13]: won't virtruKeys.data here also cause issues?

File: cks/templates/hmac-auth-secret.yaml
Comment (ehorning) [inline cks/templates/hmac-auth-secret.yaml:7]: same question here

File: cks/templates/cks-keys-secret.yaml
Comment (scjohns) [reply cks/templates/cks-keys-secret.yaml]: Oh duh. Deleted and pushed a new commit

File: cks/templates/hmac-auth-secret.yaml
Comment (scjohns) [reply cks/templates/hmac-auth-secret.yaml]: Deleted this file in new commit

File: cks/templates/configmap.yaml
Comment (scjohns) [reply cks/templates/configmap.yaml]: Yep. hardcoded for now, but I don't think this value even matters I commented it out in my CKS deployment and I can still decrypt no problem

File: cse/README.md
Comment (ehorning) [inline cse/README.md:54]: are we doing values.yaml instead of files here in anticipation of getting this into marketplace? (if so -- might need a new solution to pull in this chart in the same way as the others so probably not really necessary to plan around that here)

not going to block on it either way as I know we've talked about secrets ad-nauseum at this point and you got the go-ahead from sean/ben too, just pointing out that we don't need values.yaml here as this won't touch marketplace

**Review by ehorning — CHANGES_REQUESTED** [review]
looks great -- just a few more secrets-related comments. Also, can you [helm package](https://helm.sh/docs/helm/helm_package/) the new chart versions and add those to the repo in addition to the existing ones (e.g. https://github.com/virtru-corp/virtru-charts/blob/main/cks-0.6.1.tgz)?

File: cse/README.md
Comment (scjohns) [reply cse/README.md:54]: Wait so this chart for CSE isn't going to touch marketplace at all? 🤔  Happy change it to external secret creation if so, I just thought we were keeping CSE as is because it pushes to the marketplace deployment

**Comment by scjohns** [comment]
@ehorning packaged the charts and added them. Do I need to .gitignore them or anything?

File: cse/README.md
Comment (ehorning) [reply cse/README.md:54]: Oh sorry my bad -- for some reason I thought I was looking at gateway not cse

**Review by ehorning — DISMISSED** [review]
approving with the request that we add back in previous versions of packaged charts (tgz files) so that we can still access the previous working versions if we need to in marketplace, and also keep the ability to enable/disable tester pods

**Comment by ehorning** [comment]
> @ehorning packaged the charts and added them. Do I need to .gitignore them or anything?

No, they need to be added to git

**Review by ehorning — APPROVED** [review]

**Pull request merged**
